logic/common/log_utils.py

```diff
@@ -0,0 +1,112 @@
+''' Utility functions for printing and logging
+
+USAGE EXAMPLE:
+
+import logic.common.utils_log as log
+    log.SetVerbosityLevel(1)
+    log.Info("Hello World!")
+
+# Verbosity level is usually first set in CLI file once during initialisation
+# Afterwards, you may use log.Info(s), log.Must(s), log.Extra(s) anywhere you need
+
+'''
+
+import os
+import logging
+
+#--------------------------------------------------#
+'''Logger Settings'''
+
+DEFAULT_FILENAME = 'logfile.log'
+DEFAULT_FORMAT = '%(message)s'   # Without extra indentation
+DEFAULT_FORMAT = '\t%(message)s' # With extra indentation
+DEFAULT_DIVIDER = '--------------------------------------------------'
+
+FOLDER_LOG = 'reference/log/'
+
+DEBUG_1 = False    # Prints messages in SetVerbosityLevel() 
+
+
+
+#--------------------------------------------------#
+'''Configuration'''
+
+def SetVerbosityLevel(log_lv_num, file_name = DEFAULT_FILENAME, format = DEFAULT_FORMAT ):
+    '''Set basic configuration to logger'''
+    log_level = logging.WARNING # Default to Warning
+    if log_lv_num == 1: log_level = logging.INFO
+    elif log_lv_num == 2: log_level = logging.DEBUG
+
+    # Create a custom logger
+    logger = logging.getLogger()
+    logger.setLevel(log_level) # Set the minimum logging level
+
+    # Create a file handler to log messages to a file
+#    file_handler = logging.FileHandler(FOLDER_LOG+file_name)
+    file_handler = logging.FileHandler(FOLDER_LOG+file_name, 'w')
+    file_handler.setLevel(log_level)
+    file_handler.setFormatter(logging.Formatter(format))
+
+    # Create a stream handler for console output
+    console_handler = logging.StreamHandler()
+    console_handler.setLevel(log_level)
+    console_handler.setFormatter(logging.Formatter(format))
+
+    # Add the handlers to the logger
+    logger.addHandler(file_handler)
+    logger.addHandler(console_handler)
+
+    # Debug
+    _LoggingConfigMessage(log_lv_num)
+    if not DEBUG_1: return
+    Must("Message 0")
+    Info("Message 1")
+    Extra("Message 2")
+
+def _LoggingConfigMessage(log_lv_num):
+    print_msg = ""
+    print_msg += "\n" + DEFAULT_DIVIDER + "\n"    # Divider before meta data
+    print_msg += "  "                             # Indentation
+    print_msg += os.getcwd() + "\n"               #  Current OS
+    print_msg += "  "                             # Indentation
+    print_msg += "Verbosity = " + str(log_lv_num) #  Verbosity Level
+    print_msg += "" # TODO DateTime
+    print_msg += ""
+    print_msg += "\n" + DEFAULT_DIVIDER           # Divider between meta data & tool content
+    logging.warning(print_msg)
```

Comment:
Oh wait, looking at this again, _LoggingConfigMessage is about creating a header? (got confused since it is below the #DEBUG comment (line 59). Headers for log files are good. Let's adjust the name to reflect that. "_WriteHeaderToLog"

My other comments about removing debug stuff still applies (so lines 59, 61-64)